src/cmd/services/m3aggregator/serve/serve.go

```diff
@@ -51,7 +66,14 @@ func Serve(
 		if err := m3msgServer.ListenAndServe(); err != nil {
 			return fmt.Errorf("could not start m3msg server at: addr=%s, err=%v", m3msgAddr, err)
 		}
-		defer m3msgServer.Close()
+
+		defer func() {
```

Comment:
You could probably add all of this logging into `x/server` server.go `Close` func so you don't need to do these custom defer func

Reply:
Good idea

Reply:
Hm actually not all of these are using [this x/server](https://github.com/m3db/m3/blob/master/src/x/server/server.go#L100), e.g. the http handler just implements `x/server` [here](https://github.com/m3db/m3/blob/master/src/aggregator/server/http/server.go#L49), so this may not be the best approach